mayastor/src/bdev/dev/nvmx/channel.rs

```diff
@@ -134,54 +250,43 @@ impl NvmeIoChannelInner<'_> {
             return -libc::ENODEV;
         }
 
-        // Create qpair for target controller.
-        let mut opts = spdk_nvme_io_qpair_opts::default();
-        let default_opts = NvmeBdevOpts::default();
-
-        unsafe {
-            spdk_nvme_ctrlr_get_default_io_qpair_opts(
-                ctrlr_handle,
-                &mut opts,
-                size_of::<spdk_nvme_io_qpair_opts>() as u64,
-            );
-
-            opts.io_queue_requests =
-                max(opts.io_queue_requests, default_opts.io_queue_requests);
-            opts.create_only = true;
-
-            let qpair: *mut spdk_nvme_qpair = spdk_nvme_ctrlr_alloc_io_qpair(
-                ctrlr_handle,
-                &opts,
-                size_of::<spdk_nvme_io_qpair_opts>() as u64,
+        // We assume that channel is reinitialized after being reset, so we
+        // expect to see no I/O qpair.
+        if self.qpair.is_some() {
+            warn!(
+                ?ctrlr_name,
+                "I/O channel has active I/O qpair while being reinitialized, clearing"
             );
+            self.qpair.take().unwrap();
```

Comment:
hmm could we avoid unwrap by using like: if let Some(q) = q.take() ?

Reply:
Good point.